src/PlusOne.java

```diff
@@ -0,0 +1,20 @@
+public class PlusOne {
+    public static int[] plusOne(int[] digits) {
+        int[] result = new int[digits.length + 1];
+        result[0] = 1;
+        digits[digits.length - 1]++;
+        for(int i = digits.length - 1; i >= 0; i--) {
+            if(digits[i] > 9 && i != 0) {
+                digits[i] = 0;
+                digits[i - 1]++;
+            } else if(digits[i] > 9 && i == 0){
+                digits[i] = 0;
+            } else {
+                result = digits;
```

Comment:
I find this code confusing and difficult to follow. You allocate a new array, sometimes you return it, sometimes you don't. 
Can you change your code so that it does not modify digits array and always returns result array?